python/ray/tests/test_memory_deadlock.py

```diff
@@ -13,22 +14,26 @@
     memory_monitor_refresh_ms,
 )
 
+LIFO_POLICY = "retriable_lifo"
+DEPTH_POLICY = "group_by_depth"
 
-@pytest.fixture
-def ray_with_memory_monitor(shutdown_only):
+
+@pytest.fixture(params=[LIFO_POLICY, DEPTH_POLICY])
+def ray_with_memory_monitor(request):
     with ray.init(
-        num_cpus=1,
+        address="local",
```

Comment:
I've left this as is for now, since num_cpus=1 causes each instance of `two_sets_of_` cases to not run concurrently, thus breaking the tests. Can discuss more